chainer/distributions/dirichlet.py

```diff
@@ -0,0 +1,114 @@
+import chainer
+from chainer.backends import cuda
+from chainer import distribution
+from chainer.functions.array import broadcast
+from chainer.functions.array import expand_dims
+from chainer.functions.array import repeat
+from chainer.functions.math import digamma
+from chainer.functions.math import exponential
+from chainer.functions.math import lgamma
+from chainer.functions.math import sum as sum_mod
+import numpy
+
+
+class Dirichlet(distribution.Distribution):
+
+    """Dirichlet Distribution.
+
+    The probability density function of the distribution is expressed as
+
+    .. math::
+        p(x) = \\frac{\\Gamma(\\sum_{i=1}^{K} \\alpha_i)}
+            {\\prod_{i=1}^{K} \\Gamma (\\alpha_i)}
+            \\prod_{i=1}^{K} {x_i}^{\\alpha_i-1}
+
+    Args:
+        alpha(:class:`~chainer.Variable` or :class:`numpy.ndarray` or \
+        :class:`cupy.ndarray`): Parameter of distribution.
+    """
+
+    def __init__(self, alpha):
+        self.__alpha = chainer.as_variable(alpha)
+        self.__k = alpha.shape[-1]
+
+    @property
+    def alpha(self):
+        return self.__alpha
+
+    @property
+    def k(self):
+        return self.__k
+
+    @property
+    def alpha0(self):
+        return sum_mod.sum(self.alpha, axis=-1)
+
+    @property
+    def batch_shape(self):
+        return self.alpha.shape[:-1]
+
+    @property
+    def entropy(self):
+        return sum_mod.sum(lgamma.lgamma(self.alpha), axis=-1) \
+            - lgamma.lgamma(self.alpha0) \
```

Comment:
The multivariate log beta function (not necessarily a `FunctionNode`) could be useful in `entropy`, `log_prob`, and kl.